src/projectDB/projectversioning.h

```diff
@@ -0,0 +1,208 @@
+#ifndef PROJECTVERSIONING_H
+#define PROJECTVERSIONING_H
+
+#include <string>
+#include <map>
+
+using namespace std;
+
+namespace ProjectVersioning {
+
+/**
+ * @brief The Version struct is used to represent a version number having
+ * major, minor and patch levels (ref: htttps://semver.org). The most obvious
+ * reason for the existence of this struct is to allow comparison between two
+ * Version objects.
+ */
+struct Version {
+
+    /**
+     * @brief Create a version object from a version string of the form
+     * "MAJOR.MINOR.PATCH". If the string is not of this form, the returned
+     * Version object will most likely not get an incorrect representation than
```

Comment:
Do you mean "..will most likely get an incorrect representation" ?

Reply:
Yes I meant that. Sorry about that. Corrected it.